charmhub/refresh.go

```diff
@@ -296,7 +321,11 @@ func InstallOneFromChannel(name string, channel string, base RefreshBase) (Refre
 	if name == "" {
 		return nil, logAndReturnError(errors.NotValidf("empty name"))
 	}
-	if err := validateBase(base); err != nil {
+	base, err := base.sanatiseBase()
```

Comment:
Needed in DownloadOneFromChannel, DownloadOneFromChannelByName too. It might be better to put this logic in: constructRefreshBase() instead. Then a sanitise method wouldn't be needed.

Reply:
Yes good idea